receiver/awscontainerinsightreceiver/internal/cadvisor/cadvisor_linux.go

```diff
@@ -76,18 +76,35 @@ var defaultCreateManager = func(memoryCache *memory.InMemoryCache, sysfs sysfs.S
 	return manager.New(memoryCache, sysfs, houskeepingConfig, includedMetricsSet, collectorHTTPClient, rawContainerCgroupPathPrefixWhiteList, perfEventsFile)
 }
 
-type cadvisorOption func(*Cadvisor)
+// Option is a function that can be used to configure Cadvisor struct
+type Option func(*Cadvisor)
 
-func cadvisorManagerCreator(f createCadvisorManager) cadvisorOption {
+func cadvisorManagerCreator(f createCadvisorManager) Option {
 	return func(c *Cadvisor) {
 		c.createCadvisorManager = f
 	}
 }
 
+// DecoratorOption constructs an option for configuring the metric decorator
+func DecoratorOption(d decorator) Option {
```

Comment:
```suggestion
func WithDecorator(d decorator) Option {
```

It is idiomatic to name `Option` creation functions that provide a configuration value `WithX`.

Reply:
I will make the change.